src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -68,24 +72,29 @@ pub struct S3MetaData<S: S3BackEnd + Send + Sync + 'static> {
 }
 
 /// Parse S3 info
-fn parse_s3_info(info: &str) -> (&str, &str, &str, &str) {
+fn parse_s3_info(info: &str) -> DatenLordResult<(&str, &str, &str, &str)> {
     info.split(S3_INFO_DELIMITER)
         .next_tuple()
-        .unwrap_or_else(|| panic!("parse s3 information failed. s3_info: {info}"))
+        .ok_or_else(|| anyhow::anyhow!("parse s3 information failed. s3_info: {info}").into())
 }
 
 #[async_trait]
 impl<S: S3BackEnd + Sync + Send + 'static> MetaData for S3MetaData<S> {
     type N = S3Node<S>;
 
     #[instrument(skip(self))]
-    async fn release(&self, ino: u64, fh: u64, _flags: u32, _lock_owner: u64, flush: bool) {
-        let mut inode = self.get_node_from_kv_engine(ino).await.unwrap_or_else(|| {
-            panic!(
-                "release() found fs is inconsistent, \
-                     the inode ino={ino} is not in cache"
-            );
-        });
+    async fn release(
+        &self,
+        ino: u64,
+        fh: u64,
+        _flags: u32,
+        _lock_owner: u64,
+        flush: bool,
+    ) -> DatenLordResult<()> {
+        let mut inode = self
+            .get_node_from_kv_engine(ino)
+            .await?
+            .ok_or_else(|| error::build_inconsistent_fs(ino, "release"))?;
```

Comment:
Maybe try https://stackoverflow.com/questions/38088067/equivalent-of-func-or-function-in-rust this way to get the function name.

Reply:
Thank you for your suggestion. It's applied in my code now.